tinker/tinker_controller.py

```diff
@@ -373,21 +373,28 @@ def inspect_child(self, child, find_all):
         pass
 
     def traverse_xml(self, xml_url, type_to_find, find_all=False):
-        response = requests.get(xml_url)
-        form_xml = ET.fromstring(response.content)
+        username = session['username']
 
-        matches = []
+        # Username is used for caching purposes
+        @cache.memoize(timeout=600)
```

Comment:
can you also reduce this time to something like 5minutes? (300). I would be tempted to even reduce it a little more - so the faculty bio one is more up to date.

Reply:
Done. I stuck at 300 for now. If you decide more firmly that you want it lower just let me know!